plugins/inputs/mikrotik/mikrotik.go

```diff
@@ -0,0 +1,224 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package mikrotik
+
+import (
+	_ "embed"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf/plugins/inputs"
+
+	common_tls "github.com/influxdata/telegraf/plugins/common/tls"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Mikrotik struct {
+	Address         string          `toml:"address"`
+	IgnoreCert      bool            `toml:"ignore_cert,omitempty"`
+	ResponseTimeout config.Duration `toml:"response_timeout"`
+	IgnoreComments  []string        `toml:"ignore_comments"`
+	IncludeModules  []string        `toml:"include_modules"`
+	Username        config.Secret   `toml:"username"`
+	Password        config.Secret   `toml:"password"`
+	Log             telegraf.Logger `toml:"-"`
+	common_tls.ClientConfig
```

Comment:
Probably want to use client config from `plugins/common/http`?
```suggestion
	http.HTTPClientConfig
```